src/net/dash/isobmff_timing_infos.ts

```diff
@@ -96,18 +98,21 @@ function getISOBMFFTimingInfos(
   }
 
   if (startTime == null) {
-      if (_sidxSegments.length === 0) {
-        startTime = segmentStart;
+    if (_sidxSegments.length === 0) {
+      startTime = segmentStart;
+    } else {
+      const sidxStart = _sidxSegments[0].time;
+      if (sidxStart >= 0) {
+        const sidxTimescale = _sidxSegments[0].timescale;
+        const baseStartTime = sidxTimescale != null && sidxTimescale !== timescale ?
+          (sidxStart / sidxTimescale) * timescale : sidxStart;
+        startTime = segment.timestampOffset != null ?
+          baseStartTime + (segment.timestampOffset * timescale) :
+          baseStartTime;
       } else {
-        const sidxStart = _sidxSegments[0].time;
-        if (sidxStart >= 0) {
-          const sidxTimescale = _sidxSegments[0].timescale;
-          startTime = sidxTimescale != null && sidxTimescale !== timescale ?
-            (sidxStart / sidxTimescale) * timescale : sidxStart;
-        } else {
-          startTime = segmentStart;
-        }
+        startTime = segmentStart;
```

Comment:
Can sidxStart really can be under 0 ? And shouldn't we also offset the startTime if so ?